applications/CompressiblePotentialFlowApplication/python_scripts/define_wake_process_2d.py

```diff
@@ -0,0 +1,231 @@
+import KratosMultiphysics
+import KratosMultiphysics.CompressiblePotentialFlowApplication as CPFApp
+import math
+
+
+def Factory(settings, Model):
+    if(not isinstance(settings, KratosMultiphysics.Parameters)):
+        raise Exception(
+            "expected input shall be a Parameters object, encapsulating a json string")
+
+    return DefineWakeProcess(Model, settings["Parameters"])
+
+
+class DefineWakeProcess(KratosMultiphysics.Process):
+    def __init__(self, Model, settings):
+        KratosMultiphysics.Process.__init__(self)
+
+        default_settings = KratosMultiphysics.Parameters("""
+            {
+                "mesh_id"                   : 0,
+                "model_part_name"           : "please specify the model part that contains the kutta nodes",
+                "upper_surface_model_part_name" : "please specify the model part that contains the upper surface nodes",
+                "lower_surface_model_part_name" : "please specify the model part that contains the lower surface nodes",
+                "fluid_part_name"           : "MainModelPart",
+                "wake_direction"                 : [1.0,0.0,0.0],
+                "epsilon"    : 1e-9
+            }
+            """)
+
+        settings.ValidateAndAssignDefaults(default_settings)
+        # TODO Implement this process in C++ and make it open mp parallel to save time selecting the wake elements
```

Comment:
You are totally right. I guess, among other things, the first improvement would be to implement the process in c++ and make the loops parallel. However, I do not know whether this is a priority right now, because this process is only for 2D and the complexity of 2D problems is not very large.